subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/algorithm/hived_algorithm.go

```diff
@@ -855,6 +1139,15 @@ func (h *HivedAlgorithm) findPhysicalGpuInChain(
 	return nil
 }
 
+func nodeHealthy(node *core.Node) bool {
+	for _, c := range node.Status.Conditions {
+		if c.Type == "Ready" && c.Status != "Unknown" {
```

Comment:
>if c.Type == "Ready" && c.Status != "Unknown" { [](start = 2, length = 47)

Investigate K8S internal how to check node health?
They must not use these strings, at least is enum.
And they may also have existing health func for you to call.